### PR TITLE
Added dockerfile to allow deploying via docker

### PR DESCRIPTION
**Pull request by AnnoyingRain5**

### CODE REVIEW

File: dockerfile [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+EXPOSE 8080/tcp
+RUN apk update
+RUN apk add --no-cache npm python3 make g++
+WORKDIR /app
+COPY . .
+RUN mkdir /config
+VOLUME /config
+RUN mv example.config.json /config/config.json
+RUN npm install
+WORKDIR /app
```
Comment (mrjvs) [inline dockerfile:12]: duplicate workdir

File: dockerfile [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+RUN echo "cp /config/config.json /app/config.json &&  npm start" > startup.sh
+RUN chmod +x startup.sh
+ENTRYPOINT "/app/startup.sh"
```
Comment (mrjvs) [inline dockerfile:15]: No endline at file

File: dockerfile [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# syntax=docker/dockerfile:1
+FROM node:current-alpine3.11
+EXPOSE 8080/tcp
```
Comment (mrjvs) [inline dockerfile:3]: after adding environment variable support for config, make sure the app always run on 8080 using the env

**Review by mrjvs — CHANGES_REQUESTED** [review]
Hey, thanks for the PR.

There are a few things needed before this can be approved:
 1. Proper configuration, best practice for dockerfiles is to have configuration be set through environment if possible. in this case it needs some extra code but it is possible. So every config key needs to be able to be setup through environment variables
 2. Static image version, always using current is not recommended as it may break just by inactiveness. so dont use ` current` in the image.
 3. proper image layers:
     - Do dependencies before copying source, this will make sure dependencies are only downloaded when needed.
     - No unnecesary steps, we don't need an entrypoint, just do a `CMD node .`
     - Do workdir at the top below FROM
 4. Comment the dockerfile, we need to know whats going on :P
 5. Add a .dockerignore file, right now it will add the config.json and entire node_modules to the context when building

Also check review comments

**Review by mrjvs — APPROVED** [review]

**Comment by mrjvs** [comment]
@jonbarrow tested dockerfile, works and looks good. ready for merge

**Comment by MatthewL246** [comment]
Closing because this server has now been Dockerized in #9.

**Pull request closed without merging**